DataStorageEngine/DSECommonSenseProxy.h

```diff
@@ -0,0 +1,172 @@
+
+
+#pragma mark Device (DSESource backend equivalent) functions
+/**
+ @name Sensors and Devices (DSESource backend equivalent)
+*/
+
+/** 
+ Get all devices for the currently logged in user.
+ 
+ This will fetch all the devices for the current user and pass them on to the success callback as an NSArray. Each element in the array will contain an NSDictionary with data from one device.
+ 
+ @param successCallback A callback method that will be called in the case the API call succeeds. It receives the resulting array of devices.
+ @param failureCallback A callback method that will be called in case of failing the API call. It will receive an NSError object with details about the cause of the failure.
+ */
++ (void) getDevicesWithSuccessCallback:(void(^)(NSArray* devices)) successCallback andFailureCallback:(void(^)(NSError* error)) failureCallback;
+
+
+/**
+ Get all sensors for the currently logged in user.
+ 
+ This will fetch all the sensors for the current user and pass them on to the success callback as an NSArray. Each element in the array will contain an NSDictionary with data from one sensor.
+ 
+ @param successCallback A callback method that will be called in the case the API call succeeds. It receives the resulting array of sensors.
+ @param failureCallback A callback method that will be called in case of failing the API call. It will receive an NSError object with details about the cause of the failure.
+ */
++ (void) getSensorsWithSuccessCallback:(void(^)(NSArray* sensors)) successCallback andFailureCallback:(void(^)(NSError* error)) failureCallback;
+
+
+/** 
+ Create a new sensor in the commonsense backend
+ 
+ Each sensor in commonsense is uniquely identified by a name and devicetype combination. If this combination already exists in the commonsense backend this call will fail.
+ 
+ @param name			Name of the sensor to create. This will be used to identify the sensor.
+ @param displayName		Extra field to make sensor name more readable if necessary when displaying it.
+ @param deviceType		Name of the device type the sensor belongs to.
+ @param dataType		Type of data that the sensor stores.
+ @param dataStructure	Structure of the data in the data; can be used to specify the JSON structure in the sensor.
+ @param successCallback A callback method that will be called in the case the API call succeeds. It does not receive any input parameters.
+ @param failureCallback A callback method that will be called in case of failing the API call. It will receive an NSError object with details about the cause of the failure.
+ */
++ (void) createSensorWithName: (NSString *) name andDisplayName: (NSString *) displayName andDeviceType: (NSString *) deviceType andDataType: (NSString *) dataType andDataStructure: (NSString *) dataStructure andSuccessCallback:(void(^)(NSArray* sensors)) successCallback andFailureCallback:(void(^)(NSError* error)) failureCallback;
+
+
+//Add sensor to a device; creates a device if it doesnt exist yet; Do we need these?
```

Comment:
We need these because the createSensor method does not support device. The device of the sensor will be unspecified otherwise.